agent/utils.mjs

```diff
@@ -0,0 +1,659 @@
+        let c = s.charCodeAt( ci );
+        if( c < 128 ) {
+            arrBytes[i++] = c;
+            continue;
+        }
+        if( c < 2048 )
+            arrBytes[i++] = c >> 6 | 192; else {
+            if( c > 0xd7ff && c < 0xdc00 ) {
+                if( ++ci >= s.length )
+                    throw new Error( "UTF-8 encode: incomplete surrogate pair" );
+                const c2 = s.charCodeAt( ci );
+                if( c2 < 0xdc00 || c2 > 0xdfff ) {
+                    throw new Error(
+                        "UTF-8 encode: second surrogate character 0x" +
+                        c2.toString( 16 ) + " at index " + ci + " out of range"
+                    );
+                }
+                c = 0x10000 + ( ( c & 0x03ff ) << 10 ) + ( c2 & 0x03ff );
+                arrBytes[i++] = c >> 18 | 240;
+                arrBytes[i++] = c >> 12 & 63 | 128;
+            } else
+                arrBytes[i++] = c >> 12 | 224;
+            arrBytes[i++] = c >> 6 & 63 | 128;
+        }
+        arrBytes[i++] = c & 63 | 128;
+    }
+    return arrBytes.subarray( 0, i );
+}
+
+export function decodeUTF8( arrBytes ) {
+    // un-marshals a string from an Uint8Array,
+    // see https://gist.github.com/pascaldekloe/62546103a1576803dade9269ccf76330
+    let i = 0, s = "";
+    while( i < arrBytes.length ) {
+        let c = arrBytes[i++];
+        if( c > 127 ) {
+            if( c > 191 && c < 224 ) {
+                if( i >= arrBytes.length )
+                    throw new Error( "UTF-8 decode: incomplete 2-byte sequence" );
+                c = ( c & 31 ) << 6 | arrBytes[i++] & 63;
+            } else if( c > 223 && c < 240 ) {
+                if( i + 1 >= arrBytes.length )
+                    throw new Error( "UTF-8 decode: incomplete 3-byte sequence" );
+                c = ( c & 15 ) << 12 | ( arrBytes[i++] & 63 ) << 6 | arrBytes[i++] & 63;
+            } else if( c > 239 && c < 248 ) {
+                if( i + 2 >= arrBytes.length )
+                    throw new Error( "UTF-8 decode: incomplete 4-byte sequence" );
+                c = ( c & 7 ) <<
+                    18 | ( arrBytes[i++] & 63 ) << 12 |
+                    ( arrBytes[i++] & 63 ) << 6 | arrBytes[i++] & 63;
+            } else {
+                throw new Error(
+                    "UTF-8 decode: unknown multi-byte start 0x" +
+                    c.toString( 16 ) + " at index " + ( i - 1 )
+                );
+            }
+        }
+        if( c <= 0xffff )
+            s += String.fromCharCode( c ); else if( c <= 0x10ffff ) {
+            c -= 0x10000;
+            s += String.fromCharCode( c >> 10 | 0xd800 );
+            s += String.fromCharCode( c & 0x3FF | 0xdc00 );
+        } else {
+            throw new Error(
+                "UTF-8 decode: code point 0x" +
+                c.toString( 16 ) + " exceeds UTF-16 reach" );
+        }
+    }
+    return s;
+}
+
+export function hexToBytes( strHex, isInversiveOrder ) { // convert a hex string to a byte array
```

Comment:
No, I cannot. Because:
1) `crypto-js` offers worst possible documentation.
2) `hexToBytes` supports `isInversiveOrder` parameter when `crypto-js` is not.